bootstrap/util/configowner.go

```diff
@@ -75,7 +75,7 @@ func (co ConfigOwner) IsControlPlaneMachine() bool {
 // GeConfigOwner returns the Unstructured object owning the current resource.
 func GetConfigOwner(ctx context.Context, c client.Client, obj metav1.ObjectMeta) (*ConfigOwner, error) {
 	for _, ref := range obj.OwnerReferences {
-		if ref.Kind == "Machine" && ref.APIVersion == clusterv1.GroupVersion.String() {
+		if (ref.Kind == "Machine" || ref.Kind == "MachinePool") && ref.APIVersion == clusterv1.GroupVersion.String() {
```

Comment:
This change lgtm, but can you also add additional tests for this behavior in `bootstra/util/configowner_test.go`?

Reply:
Shouldn't we also adjust it to ignore the version portion of ref.APIVersion and just compare on Group + Kind?

Reply:
Going to merge this and open another PR to follow-up with these comments